extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-base/src/main/java/org/apache/guacamole/auth/jdbc/JDBCAuthenticationProviderService.java

```diff
@@ -159,13 +182,21 @@ else if (!databaseRestrictionsApplicable) {
 
     }
 
+    @Override
+    public ModeledUserContext getUserContext(AuthenticationProvider authenticationProvider,
+            AuthenticatedUser authenticatedUser) throws GuacamoleException {
+
+        // Do not force refresh unless updateUserContext is explicitly called
+        return getUserContext(authenticationProvider, authenticatedUser, false);
+    }
+
     @Override
     public UserContext updateUserContext(AuthenticationProvider authenticationProvider,
             UserContext context, AuthenticatedUser authenticatedUser,
             Credentials credentials) throws GuacamoleException {
 
-        // No need to update the context
-        return context;
+        // Force-refresh the user context
+        return getUserContext(authenticationProvider, authenticatedUser, true);
```

Comment:
This is to ensure that the KSM config value saved on the user preferences screen is up to date. Without this, you'd have to log out and back in again before the correct value would be pulled (when doing token replacement on a connection).

Reply:
Is there a need for `forceRefresh`? `getUserContext()` of `AuthenticationProvider` is called at most once, and only during initial auth, with `updateUserContext()` being the function that's called for established sessions.

Reply:
Ah, I see. In that case it seems like the `authenticatedUser instanceof ModeledAuthenticatedUser` check in `retrieveUser` isn't really needed, yeah? I'll remove the flag and simplify.

Reply:
> Ah, I see. In that case it seems like the `authenticatedUser instanceof ModeledAuthenticatedUser` check in `retrieveUser` isn't really needed, yeah?

Well, _that_ check is needed as `authenticatedUser` is not guaranteed to be an implementation of `AuthenticatedUser` from this extension (`ModeledAuthenticatedUser`). If the user is authenticated by some other extension, such as LDAP or SAML, it'll be the `AuthenticatedUser` implementation from that extension.

Reply:
I still don't get why that's needed, though. I mean, getting rid of that check just forces it to reload from the DB every time, which is what we want.

That check is what `forceRefresh` existed to specifically disable, which is what's needed to get the user context to actually refresh when `updateUserContext` is called.

Reply:
You're right - I was thinking about acceptance of `AuthenticatedUser` objects from other extensions, which has no impact here except if possibly reading the cached user object, which we no longer need to do.